src/python/pants/help/help_printer.py

```diff
@@ -56,7 +57,17 @@ def print_hint() -> None:
         elif isinstance(self._help_request, OptionsHelp):
             self._print_options_help()
         elif isinstance(self._help_request, UnknownGoalHelp):
```

Comment:
On the face of it, I would expect the other invocation of `difflib` to also be in this file, in response to an `UnknownFlagsHelp` help request.

Reply:
Yeah, it's slightly quirky, but this file is for help requests (where we could determine that we need to show help at arg splitting time), whereas since we parse flags ~lazily, we can't fit it neatly into that framework.